components/provisioner/internal/installation/resource_cleanup.go

```diff
@@ -46,86 +46,139 @@ type serviceCatalogClient struct {
 }
 
 func (s *serviceCatalogClient) PerformCleanup() error {
-	//1. Fetch all service cluster service brokers
+	// Fetch all ClusterServiceBrokers
+	//TODO: Add retries for listing resources in case api-server fails to respond
 	clusterServiceBrokers, err := s.ListClusterServiceBroker(metav1.ListOptions{})
 	if err != nil {
-		return errors.Wrapf(err, "while listing cluster service brokers")
+		return errors.Wrapf(err, "while listing ClusterServiceBrokers")
 	}
 
-	//2. Filter based on broker URL prefix
+	// Filter the list based on ClusterServiceBroker's URL prefix
+	brokersWithUrlPrefix := s.FilterCsbWithUrlPrefix(clusterServiceBrokers, BrokerUrlPrefix)
+
+	// Get all ClusterServiceClasses from filtered ClusterServiceBrokers
```

Comment:
I'm not convinced for using comments in such a way. What we do here is repeating the same information ; function names along with variable names provide enough context in this case.